### PR TITLE
Mark up `playlist/ShareModal` for translations

### LINKED ISSUE #73619: i18n: Mark up `playlist/ShareModal` for translations
playlist share modal is accessed by pressing the share button on an existing playlist: 
image
image

follow instructions [here](https://github.com/grafana/grafana/blob/main/contribute/internationalization.md#how-to-add-a-new-translation-phrase). can verify translation strings have been marked up correctly by using the development-only pseudo locale: 
image

checklist:
- [ ] modal title
- [ ] close tooltip
- [ ] all other text in the modal **except** the url itself

### PR DESCRIPTION
**Pull request by aditipatelpro**
**What is this feature?**

Adds markup for translation in playlist/shareModal.

**Why do we need this feature?**

To ensure all string literals are translated properly.

**Who is this feature for?**

Everyone

**Which issue(s) does this PR fix?**:

Fixes #73619 

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Comment by aditipatelpro** [comment]
Hi @ashharrison90, I have completed working on the issue #73619. Could you please provide feedback on it?

File: packages/grafana-ui/src/components/Modal/Modal.tsx [hunk 1]
```diff
@@ -4,6 +4,8 @@ import { FocusScope } from '@react-aria/focus';
 import { OverlayContainer, useOverlay } from '@react-aria/overlays';
 import React, { PropsWithChildren, useRef } from 'react';
 
+import { t } from 'app/core/internationalization';
```
Comment (ashharrison90) [inline packages/grafana-ui/src/components/Modal/Modal.tsx:7]:
```suggestion
import { t } from '../../utils/i18n';
```

in grafana-ui, we need to import the utils from a different place as per https://github.com/grafana/grafana/blob/main/contribute/internationalization.md#how-to-add-a-new-translation-phrase:~:text=When%20translating%20in%20grafana%2Dui%2C%20use%20a%20relative%20path%20to%20import%20%3CTrans%20/%3E%20and%20t()%20from%20src/utils/i18n.

File: public/app/features/playlist/ShareModal.tsx [hunk 2]
```diff
@@ -16,9 +17,9 @@ export const ShareModal = ({ playlistUid, onDismiss }: Props) => {
   const [autoFit, setAutofit] = useState(false);
 
   const modes: Array<SelectableValue<PlaylistMode>> = [
-    { label: 'Normal', value: false },
-    { label: 'TV', value: 'tv' },
-    { label: 'Kiosk', value: true },
+    { label: t('share-modal.playlist.mode-normal', 'Normal'), value: false },
```
Comment (ashharrison90) [inline public/app/features/playlist/ShareModal.tsx:20]: `share-modal` refers to a different component. can we instead just create a new prefix to represent this different component, `share-playlist`? so the keys would be like `share-playlist.mode-normal`, `share-playlist.title`, `share-playlist.copy-link-button`, etc.

**Review by ashharrison90 — COMMENTED** [review]
couple of minor tweaks, otherwise looks great 🙌

File: public/app/features/playlist/ShareModal.tsx (hunk 2 above)
Comment (aditipatelpro) [reply public/app/features/playlist/ShareModal.tsx:20]: Sure, I'll make the changes accordingly.

**Comment by aditipatelpro** [comment]
Hi @ashharrison90 , I have completed working on your feedback. Could you please review it?

**Review by ashharrison90 — APPROVED** [review]
looks great! 🥳 

thanks so much for your contribution to grafana @aditipatelpro 🙇

**Pull request merged**